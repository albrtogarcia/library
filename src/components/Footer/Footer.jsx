import { Link } from 'react-router-dom'
import './Footer.scss'

export const Footer = () => {
  const current = new Date()
  const year = current.getFullYear()
  const syncDate = current.toLocaleString('es-ES', {
    day: 'numeric',
    month: 'numeric',
    year: 'numeric',
    hour: 'numeric',
    minute: 'numeric'
  })

  return (
    <footer className="footer">
      <div className="copy">
        <small>&copy;{year} Movistar Plus+</small>
      </div>

      <div className="footer-nav">
        <a href="#" className="nav-item">
          <span>Aviso legal</span>
        </a>
        <a href="#" className="nav-item">
          <span>Condiciones generales</span>
        </a>
        <a href="#" className="nav-item">
          <span>Privacidad</span>
        </a>
        <a href="#" className="nav-item">
          <span>Cookies</span>
        </a>
        <Link to="/documentacion" className="nav-item">
          <span>Documentación</span>
        </Link>
      </div>

      <div className="info-sync">
        <small>
          Última actualización: <time dateTime="2023-08-30T12:45">{syncDate}</time>
        </small>
      </div>
    </footer>
  )
}
