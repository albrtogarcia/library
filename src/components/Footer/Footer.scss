.footer {
  display: flex;
  padding: 1rem 2rem;
  align-items: center;
  justify-content: space-between;
  font-size: 0.75rem;
  color: var(--slate-400, #94A3B8);

  &-nav {
    display: flex;
    justify-content: center;
    gap: 1rem;
  }

  .nav-item {
    color: var(--slate-400, #94A3B8);
    font-size: 0.875rem;
  }
}