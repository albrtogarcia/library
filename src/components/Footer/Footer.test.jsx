import { describe, expect, it } from 'vitest'
import { Footer } from './Footer'

describe('Footer', () => {
  it('matches snapshot', () => {
    const element = Footer()
    expect(element).toMatchSnapshot()
  })
})
