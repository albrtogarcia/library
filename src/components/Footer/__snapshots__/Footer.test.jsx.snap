// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > matches snapshot 1`] = `
<footer
  className="footer"
>
  <div
    className="copy"
  >
    <small>
      ©
      2023
       Movistar Plus+
    </small>
  </div>
  <div
    className="footer-nav"
  >
    <a
      className="nav-item"
      href="#"
    >
      <span>
        Aviso legal
      </span>
    </a>
    <a
      className="nav-item"
      href="#"
    >
      <span>
        Condiciones generales
      </span>
    </a>
    <a
      className="nav-item"
      href="#"
    >
      <span>
        Privacidad
      </span>
    </a>
    <a
      className="nav-item"
      href="#"
    >
      <span>
        Cookies
      </span>
    </a>
    <Link
      className="nav-item"
      to="/documentacion"
    >
      <span>
        Documentación
      </span>
    </Link>
  </div>
  <div
    className="info-sync"
  >
    <small>
      Última actualización: 
      <time
        dateTime="2023-08-30T12:45"
      >
        23/10/2023, 13:33
      </time>
    </small>
  </div>
</footer>
`;
